model_v4.py

```diff
@@ -0,0 +1,382 @@
+                              padding='same')(input_layer)
+        else:
+            raise ValueError('set pool_type max for MaxPooling2D or avg for AveragePooling2D')
+        setattr(self, f'inception_{addition_name}_pool_conv',
+                Conv2D(filters=pool_filters,
+                       kernel_size=(1, 1),
+                       padding='same',
+
+                       kernel_regularizer=l2(0.01),
+                       name=f'inception_{addition_name}_pool_conv')(mp))
+        setattr(self, f'inception_{addition_name}_pool_bn',
+                BatchNormalization(name=f'inception_{addition_name}_pool_bn')(
+                    getattr(self, f'inception_{addition_name}_pool_conv')))
+        setattr(self, f'inception_{addition_name}_pool_act',
+                Activation('relu', name=f'inception_{addition_name}_pool_act')(
+                    getattr(self, f'inception_{addition_name}_pool_bn')))
+
+        setattr(self, f'inception_{addition_name}_1x1_conv',
+                Conv2D(filters=conv_filters,
+                       kernel_size=(1, 1),
+                       padding='same',
+                       kernel_regularizer=l2(0.01),
+                       name=f'inception_{addition_name}_1x1_conv')(input_layer))
+        setattr(self, f'inception_{addition_name}_1x1_bn',
+                BatchNormalization(name=f'inception_{addition_name}_1x1_bn')
+                (getattr(self, f'inception_{addition_name}_1x1_conv')))
+        setattr(self, f'inception_{addition_name}_1x1_act',
+                Activation('relu', name=f'inception_{addition_name}_1x1_act')(
+                    getattr(self, f'inception_{addition_name}_1x1_bn')))
+
+    def create_interception_block(self, input_layer, addition_name='',
+                                    conv1_5x5=None, conv2_5x5=None,
+                                    conv1_3x3=None, conv2_3x3=None,
+                                    conv_filters=None, pool_filters=None,
+                                    pool_type='max', addition_pool=False):
+        """
+        Create inception block
+        :param input_layer: the last layer
+        :param addition_name:  index in the naming system (idx of inception layer)
+        :param conv1_5x5: num of filters in conv1_5x5 layer; None if that layer doesn't need
+        :param conv2_5x5: num of filters in conv2_5x5 layer; None if that layer doesn't need
+        :param conv1_3x3: num of filters in conv1_3x3 layer; None if that layer doesn't need
+        :param conv2_3x3: num of filters in conv2_3x3 layer; None if that layer doesn't need
+        :param conv_filters: num of filters in conv2 layer; None if that layer doesn't need
+        :param pool_filters: num of filters in conv1 layer; None if that layer doesn't need
+        :param pool_type:  'max' - use MaxPooling or 'avg' - use AveragePooling
+        :param addition_pool: add a stream with MaxPooling layer
+        :return: None
+        """
+        layers_set_list = []
```

Comment:
should be removed

Reply:
Actually, I don't understand the reason. It's used for collections of output layers for following concatenating.

Reply:
If you split this very large function, then you will be be able smth like this:
```
x1 = Conv1(X)
x2 = BN(X1)
x3_1 = Activation(x2)
...
return concatenate([x3_1, x3_2, ..])
```
So, the main reason here is to improve readability. We have well-defined operations number in the each inception block, so it's better to not use generic approach here and add references to collection.